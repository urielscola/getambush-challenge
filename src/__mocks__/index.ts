export * from './requests';
export * from './utils';
