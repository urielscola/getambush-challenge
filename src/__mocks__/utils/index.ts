export * from './withExternalDependencies';
export * from './mockApi';
