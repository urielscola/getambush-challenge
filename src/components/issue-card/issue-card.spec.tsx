import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  render,
  fireEvent,
  act,
  waitFor,
  RenderResult,
} from '@testing-library/react';
import { getIssuesSuccess, withExternalDependencies } from '__mocks__';

import IssueCard from './issue-card';

describe('IssueCard component', () => {
  it('should render', async () => {
    let result: RenderResult;

    await act(async () => {
      result = render(
        withExternalDependencies(
          <IssueCard issue={getIssuesSuccess.items[0]} index={1} />
        )
      );
    });

    await waitFor(() => {
      expect(result).toMatchSnapshot();
    });
  });

  it('should handle favorite click action', async () => {
    let result: RenderResult | any;

    await act(async () => {
      result = render(
        withExternalDependencies(
          <IssueCard issue={getIssuesSuccess.items[0]} index={1} />
        )
      );
    });

    const button = result.container.querySelector('#card-top svg');
    fireEvent.click(button);

    await waitFor(() => {
      expect(result).toMatchSnapshot();
    });
  });
});
