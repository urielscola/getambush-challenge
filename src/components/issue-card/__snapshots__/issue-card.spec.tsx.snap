// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueCard component should handle favorite click action 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-egiyK jlqOBX"
      >
        <div
          class="sc-gsDKAQ eGzpGk"
          data-testid="flex-div"
          display="flex"
          id="card-top"
        >
          <picture
            class="sc-bqiRlB hkrURW"
          >
            <img
              src="https://avatars.githubusercontent.com/u/77824565?v=4"
            />
          </picture>
          <h4
            class="sc-eCImPb gZUvNq"
            color="#fff"
            data-testid="text"
          >
            doflo-dfa
          </h4>
          <button
            aria-hidden="true"
            class="sc-hKwDye hXqDqF"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 bsimgx"
              color="#e35b5c"
              fill="currentColor"
              focusable="false"
              height="25"
              viewBox="0 0 16 16"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-gsDKAQ AnHBp"
          data-testid="flex-div"
          display="flex"
        >
          <h3
            class="sc-eCImPb fwIuPG"
            data-testid="text"
          >
            <a
              href="https://github.com/wbkd/react-flow/issues/1774"
              target="_blank"
              title="V10 API Example - Infinite loop ?"
            >
              V10 API Example - Infinite loop ?
            </a>
          </h3>
          <p
            class="sc-eCImPb hjhHUe"
            color="#33373c"
            data-testid="text"
          >
            repo: 
            react-flow
          </p>
          <div
            class="sc-gsDKAQ edTmWX"
            data-testid="flex-div"
            display="flex"
          >
            <div
              aria-hidden="true"
              class="sc-dkPtRN buaSqF"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 bsimgx"
                color="#2caa3b"
                fill="currentColor"
                focusable="false"
                height="30"
                viewBox="0 0 512 512"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <p
              class="sc-eCImPb cJrCDy"
              color="#272b30"
              data-testid="text"
            >
              State: 
              open
            </p>
          </div>
          <div
            class="sc-gsDKAQ edTmWX"
            data-testid="flex-div"
            display="flex"
          >
            <div
              aria-hidden="true"
              class="sc-dkPtRN buaSqF"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 bsimgx"
                color="#a1a5ab"
                fill="currentColor"
                focusable="false"
                height="30"
                viewBox="0 0 24 24"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 2H4c-1.103 0-2 .897-2 2v18l5.333-4H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14H6.667L4 18V4h16v12z"
                />
              </svg>
            </div>
            <p
              class="sc-eCImPb cJrCDy"
              color="#272b30"
              data-testid="text"
            >
              Comments: 
              0
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-egiyK jlqOBX"
    >
      <div
        class="sc-gsDKAQ eGzpGk"
        data-testid="flex-div"
        display="flex"
        id="card-top"
      >
        <picture
          class="sc-bqiRlB hkrURW"
        >
          <img
            src="https://avatars.githubusercontent.com/u/77824565?v=4"
          />
        </picture>
        <h4
          class="sc-eCImPb gZUvNq"
          color="#fff"
          data-testid="text"
        >
          doflo-dfa
        </h4>
        <button
          aria-hidden="true"
          class="sc-hKwDye hXqDqF"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-ea9ulj-0 bsimgx"
            color="#e35b5c"
            fill="currentColor"
            focusable="false"
            height="25"
            viewBox="0 0 16 16"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-gsDKAQ AnHBp"
        data-testid="flex-div"
        display="flex"
      >
        <h3
          class="sc-eCImPb fwIuPG"
          data-testid="text"
        >
          <a
            href="https://github.com/wbkd/react-flow/issues/1774"
            target="_blank"
            title="V10 API Example - Infinite loop ?"
          >
            V10 API Example - Infinite loop ?
          </a>
        </h3>
        <p
          class="sc-eCImPb hjhHUe"
          color="#33373c"
          data-testid="text"
        >
          repo: 
          react-flow
        </p>
        <div
          class="sc-gsDKAQ edTmWX"
          data-testid="flex-div"
          display="flex"
        >
          <div
            aria-hidden="true"
            class="sc-dkPtRN buaSqF"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 bsimgx"
              color="#2caa3b"
              fill="currentColor"
              focusable="false"
              height="30"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="sc-eCImPb cJrCDy"
            color="#272b30"
            data-testid="text"
          >
            State: 
            open
          </p>
        </div>
        <div
          class="sc-gsDKAQ edTmWX"
          data-testid="flex-div"
          display="flex"
        >
          <div
            aria-hidden="true"
            class="sc-dkPtRN buaSqF"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 bsimgx"
              color="#a1a5ab"
              fill="currentColor"
              focusable="false"
              height="30"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 2H4c-1.103 0-2 .897-2 2v18l5.333-4H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14H6.667L4 18V4h16v12z"
              />
            </svg>
          </div>
          <p
            class="sc-eCImPb cJrCDy"
            color="#272b30"
            data-testid="text"
          >
            Comments: 
            0
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`IssueCard component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-egiyK jlqOBX"
      >
        <div
          class="sc-gsDKAQ eGzpGk"
          data-testid="flex-div"
          display="flex"
          id="card-top"
        >
          <picture
            class="sc-bqiRlB hkrURW"
          >
            <img
              src="https://avatars.githubusercontent.com/u/77824565?v=4"
            />
          </picture>
          <h4
            class="sc-eCImPb gZUvNq"
            color="#fff"
            data-testid="text"
          >
            doflo-dfa
          </h4>
          <button
            aria-hidden="true"
            class="sc-hKwDye hXqDqF"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 bsimgx"
              color="#fff"
              fill="currentColor"
              focusable="false"
              height="25"
              viewBox="0 0 16 16"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-gsDKAQ AnHBp"
          data-testid="flex-div"
          display="flex"
        >
          <h3
            class="sc-eCImPb fwIuPG"
            data-testid="text"
          >
            <a
              href="https://github.com/wbkd/react-flow/issues/1774"
              target="_blank"
              title="V10 API Example - Infinite loop ?"
            >
              V10 API Example - Infinite loop ?
            </a>
          </h3>
          <p
            class="sc-eCImPb hjhHUe"
            color="#33373c"
            data-testid="text"
          >
            repo: 
            react-flow
          </p>
          <div
            class="sc-gsDKAQ edTmWX"
            data-testid="flex-div"
            display="flex"
          >
            <div
              aria-hidden="true"
              class="sc-dkPtRN buaSqF"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 bsimgx"
                color="#2caa3b"
                fill="currentColor"
                focusable="false"
                height="30"
                viewBox="0 0 512 512"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <p
              class="sc-eCImPb cJrCDy"
              color="#272b30"
              data-testid="text"
            >
              State: 
              open
            </p>
          </div>
          <div
            class="sc-gsDKAQ edTmWX"
            data-testid="flex-div"
            display="flex"
          >
            <div
              aria-hidden="true"
              class="sc-dkPtRN buaSqF"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 bsimgx"
                color="#a1a5ab"
                fill="currentColor"
                focusable="false"
                height="30"
                viewBox="0 0 24 24"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 2H4c-1.103 0-2 .897-2 2v18l5.333-4H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14H6.667L4 18V4h16v12z"
                />
              </svg>
            </div>
            <p
              class="sc-eCImPb cJrCDy"
              color="#272b30"
              data-testid="text"
            >
              Comments: 
              0
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-egiyK jlqOBX"
    >
      <div
        class="sc-gsDKAQ eGzpGk"
        data-testid="flex-div"
        display="flex"
        id="card-top"
      >
        <picture
          class="sc-bqiRlB hkrURW"
        >
          <img
            src="https://avatars.githubusercontent.com/u/77824565?v=4"
          />
        </picture>
        <h4
          class="sc-eCImPb gZUvNq"
          color="#fff"
          data-testid="text"
        >
          doflo-dfa
        </h4>
        <button
          aria-hidden="true"
          class="sc-hKwDye hXqDqF"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-ea9ulj-0 bsimgx"
            color="#fff"
            fill="currentColor"
            focusable="false"
            height="25"
            viewBox="0 0 16 16"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-gsDKAQ AnHBp"
        data-testid="flex-div"
        display="flex"
      >
        <h3
          class="sc-eCImPb fwIuPG"
          data-testid="text"
        >
          <a
            href="https://github.com/wbkd/react-flow/issues/1774"
            target="_blank"
            title="V10 API Example - Infinite loop ?"
          >
            V10 API Example - Infinite loop ?
          </a>
        </h3>
        <p
          class="sc-eCImPb hjhHUe"
          color="#33373c"
          data-testid="text"
        >
          repo: 
          react-flow
        </p>
        <div
          class="sc-gsDKAQ edTmWX"
          data-testid="flex-div"
          display="flex"
        >
          <div
            aria-hidden="true"
            class="sc-dkPtRN buaSqF"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 bsimgx"
              color="#2caa3b"
              fill="currentColor"
              focusable="false"
              height="30"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="sc-eCImPb cJrCDy"
            color="#272b30"
            data-testid="text"
          >
            State: 
            open
          </p>
        </div>
        <div
          class="sc-gsDKAQ edTmWX"
          data-testid="flex-div"
          display="flex"
        >
          <div
            aria-hidden="true"
            class="sc-dkPtRN buaSqF"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 bsimgx"
              color="#a1a5ab"
              fill="currentColor"
              focusable="false"
              height="30"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 2H4c-1.103 0-2 .897-2 2v18l5.333-4H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14H6.667L4 18V4h16v12z"
              />
            </svg>
          </div>
          <p
            class="sc-eCImPb cJrCDy"
            color="#272b30"
            data-testid="text"
          >
            Comments: 
            0
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
